gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -1430,14 +1430,21 @@ function GM:PlayerTakeDamage(ent, infl, att, amount, dmginfo)
 	-- start painting blood decals
 	util.StartBleeding(ent, dmginfo:GetDamage(), 5)
 
-	-- general actions for pvp damage
+	-- handle damage scaling by karma
 	if ent ~= att and IsValid(att) and att:IsPlayer() and GetRoundState() == ROUND_ACTIVE and math.floor(dmginfo:GetDamage()) > 0 then
-
 		-- scale everything to karma damage factor except the knife, because it
 		-- assumes a kill
 		if not dmginfo:IsDamageType(DMG_SLASH) then
 			dmginfo:ScaleDamage(att:GetDamageFactor())
 		end
+	end
+
+	-- before the karma is calculated, but after all other damage hooks / damage change is processed,
+	-- the armor system should come into place (GM functions are called last)
+	HandlePlayerArmorSystem(ent, infl, att, amount, dmginfo)
```

Comment:
Could you explain, why this is excluded from the previous if statement?

Reply:
Because I wanted to keep it a bit seperated to not clutter the code above. Additionally I like to work with `if bla then return end` instead of nested ifs